pontoon/actionlog/migrations/0004_auto_20240816_1045.py

```diff
@@ -0,0 +1,54 @@
+# Generated by Django 4.2.11 on 2024-08-16 10:45
+
+import datetime
+
+from django.db import migrations
+
+
+def migrate_translation_to_actionlog(apps, schema_editor):
+    Translation = apps.get_model("base", "Translation")
+    ActionLog = apps.get_model("actionlog", "ActionLog")
+
+    traslation_info = {
+        "date": ("translation:created", "Translation created"),
+        "approved_date": ("translation:approved", "Translation approved"),
+        "unapproved_date": ("translation:unapproved", "Translation unapproved"),
+        "rejected_date": ("translation:rejected", "Translation rejected"),
+        "unrejected_date": ("translation:unrejected", "Translation unrejected"),
+    }
+
+    # date to end migration
+    end_date = datetime.datetime(
+        2020, 1, 7, 9, 25, 11, 829125, tzinfo=datetime.timezone.utc
+    )
+
+    actions_to_log = []
+
+    for translation in Translation.objects.filter(date__lt=end_date):
+        for attr, action_type in traslation_info.items():
+            if (
+                getattr(translation, attr) is not None
+                and translation["attr"] < end_date
```

Comment:
I'd expect this line to raise a `TypeError`. Doesn't it?

Since we already get the value with `getattr(translation, attr)`, I'd move that call before the `if` clause:

```python
value = getattr(translation, attr)
if value is not None and value < end_date:
    ...
```

Reply:
on it

Reply:
> I'd expect this line to raise a `TypeError`. Doesn't it?
> 
> Since we already get the value with `getattr(translation, attr)`, I'd move that call before the `if` clause:
> 
> ```python
> value = getattr(translation, attr)
> if value is not None and value < end_date:
>     ...
> ```

thanks on this :)